src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -601,16 +630,14 @@ static int do_map()
 
   size = info.size;
 
-  if (size > (1UL << 32) * 512) {
-    r = -EFBIG;
-    cerr << "rbd-nbd: image is too large (" << prettybyte_t(size) << ", max is "
-         << prettybyte_t((1UL << 32) * 512) << ")" << std::endl;
+  r = ioctl(nbd, NBD_SET_SIZE, size);
```

Comment:
Could you explain why you remove size constraint check？ what happens if run in 32 bit system?